File: frontend/src/constants/trade.ts
export const TRADE_STATUSES = [
  { value: 'pending', label: '待确认', type: 'warning' },
  { value: 'confirmed', label: '已确认', type: 'primary' },
  { value: 'completed', label: '已完成', type: 'success' },
  { value: 'cancelled', label: '已取消', type: 'info' },
] as const

export const REVIEW_RATINGS = [
  { value: 'good', label: '好评' },
  { value: 'medium', label: '中评' },
  { value: 'bad', label: '差评' },
] as const

// 一次性面交码状态：unused 待核销 / used 已核销 / expired 已过期 / canceled 已作废
export const HANDOVER_STATUSES = [
  { value: 'unused', label: '待核销', type: 'warning' },
  { value: 'used', label: '已核销', type: 'success' },
  { value: 'expired', label: '已过期', type: 'danger' },
  { value: 'canceled', label: '已作废', type: 'info' },
] as const

export function tradeStatusLabel(value: string): string {
  return TRADE_STATUSES.find((t) => t.value === value)?.label ?? value
}

export function tradeStatusType(value: string): string {
  return TRADE_STATUSES.find((t) => t.value === value)?.type ?? 'info'
}

export function handoverStatusLabel(value: string): string {
  return HANDOVER_STATUSES.find((s) => s.value === value)?.label ?? '无面交码'
}

export function handoverStatusType(value: string): string {
  return HANDOVER_STATUSES.find((s) => s.value === value)?.type ?? 'info'
}

export function ratingLabel(value: string): string {
  return REVIEW_RATINGS.find((r) => r.value === value)?.label ?? value
}

File: backend/internal/constants/trade.go
package constants

// TradeStatus defines trade order state machine values shared with the frontend.
const (
	TradeStatusPending   = "pending"
	TradeStatusConfirmed = "confirmed"
	TradeStatusCompleted = "completed"
	TradeStatusCancelled = "cancelled"
)

// TradeStatuses lists all valid trade statuses in flow order.
var TradeStatuses = []string{
	TradeStatusPending, TradeStatusConfirmed, TradeStatusCompleted, TradeStatusCancelled,
}

// IsTradeStatus reports whether the given status is valid.
func IsTradeStatus(s string) bool {
	for _, v := range TradeStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// TradeStatusText returns the Chinese label of a trade status.
func TradeStatusText(s string) string {
	switch s {
	case TradeStatusPending:
		return "待确认"
	case TradeStatusConfirmed:
		return "已确认"
	case TradeStatusCompleted:
		return "已完成"
	case TradeStatusCancelled:
		return "已取消"
	default:
		return "未知"
	}
}

// ReviewRating defines review rating enum values shared with the frontend.
const (
	ReviewRatingGood   = "good"
	ReviewRatingMedium = "medium"
	ReviewRatingBad    = "bad"
)

// HandoverCodeStatus defines the lifecycle states of the one-time in-person
// handover code, shared with the frontend.
const (
	HandoverCodeUnused   = "unused"   // 已生成、等待卖家核销
	HandoverCodeUsed     = "used"     // 已核销（订单完成）
	HandoverCodeExpired  = "expired"  // 超过有效期
	HandoverCodeCanceled = "canceled" // 订单取消，码作废
)

// HandoverCodeStatuses lists all valid handover code statuses.
var HandoverCodeStatuses = []string{
	HandoverCodeUnused, HandoverCodeUsed, HandoverCodeExpired, HandoverCodeCanceled,
}

// IsHandoverCodeStatus reports whether the given status is valid.
func IsHandoverCodeStatus(s string) bool {
	for _, v := range HandoverCodeStatuses {
		if v == s {
			return true
		}
	}
	return false
}

// HandoverCodeStatusText returns the Chinese label of a handover code status.
func HandoverCodeStatusText(s string) string {
	switch s {
	case HandoverCodeUnused:
		return "待核销"
	case HandoverCodeUsed:
		return "已核销"
	case HandoverCodeExpired:
		return "已过期"
	case HandoverCodeCanceled:
		return "已作废"
	default:
		return "无面交码"
	}
}

// ReviewRatingText returns the Chinese label of a review rating.
func ReviewRatingText(r string) string {
	switch r {
	case ReviewRatingGood:
		return "好评"
	case ReviewRatingMedium:
		return "中评"
	case ReviewRatingBad:
		return "差评"
	default:
		return "未知"
	}
}
